core/server/proxy/src/main/java/alluxio/proxy/s3/logging/RequestLoggingFilter.java

```diff
@@ -11,30 +11,57 @@
 
 package alluxio.proxy.s3.logging;
 
+import org.glassfish.jersey.server.ContainerRequest;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
 import javax.ws.rs.ext.Provider;
 
 /**
  * Helper class for adding logs to request methods for debugging purposes.
  */
 @Logged
+@PreMatching
 @Provider
 public class RequestLoggingFilter implements ContainerRequestFilter {
   private static final Logger LOG = LoggerFactory.getLogger(RequestLoggingFilter.class);
 
   @Override
   public void filter(ContainerRequestContext context) throws IOException {
-    LOG.info("Method: " + context.getMethod());
-    LOG.info("Path: " + context.getUriInfo().getPath());
-    LOG.info("Query Parameters: " + context.getUriInfo().getQueryParameters());
-    LOG.info("Path Parameters: " + context.getUriInfo().getPathParameters());
-    LOG.info("Headers: " + context.getHeaders());
-    LOG.info("Media Type: " + context.getMediaType());
-    LOG.info("Request Message Body: " + context.getEntityStream());
+    if (LOG.isInfoEnabled()) {
```

Comment:
I'm not sure if `LOG.isInfoEnabled()` is necessary here. Also does this return `true` when the log level is `DEBUG`?

Reply:
yes, it will return `true` when the log level is `DEBUG`. The purpose of this modification is if the log level is over `INFO`, we don't need to construct the log StringBuilder Object.